Source/WebKit/NetworkProcess/NetworkProcess.cpp

```diff
@@ -404,8 +404,15 @@ void NetworkProcess::createNetworkConnectionToWebProcess(ProcessIdentifier ident
         connection->connection().setIgnoreInvalidMessageForTesting();
 #endif
 
-    if (auto* session = networkSession(sessionID))
-        session->protectedStorageManager()->startReceivingMessageFromConnection(Ref { connection->connection() });
+    if (auto* session = networkSession(sessionID)) {
+        Vector<WebCore::RegistrableDomain> allowedSites;
+        auto iter = m_allowedFirstPartiesForCookies.find(identifier);
+        if (iter != m_allowedFirstPartiesForCookies.end()) {
+            for (auto& site : iter->value.second)
+                allowedSites.append(site);
+        }
+        session->protectedStorageManager()->startReceivingMessageFromConnection(Ref { connection->connection() }, allowedSites);
```

Comment:
We should add `NetworkConnectionToWebProcess::protectedConnection()` and call it here.